src/drive/web/modules/views/OnlyOffice/Toolbar/Sharing.jsx

```diff
@@ -0,0 +1,31 @@
+import React, { useState } from 'react'
+
+import { ShareButton, ShareModal, SharedRecipients } from 'cozy-sharing'
+
+const Sharing = ({ fileWithPath }) => {
+  const [showShareModal, setShowShareModal] = useState(false)
+
+  return (
+    <>
+      <SharedRecipients
+        docId={fileWithPath.id}
+        size={32}
+        onClick={() => setShowShareModal(true)}
```

Comment:
useCallback :D

Reply:
Comme on ne peut pas utiliser useCallback conditionnellement, on ne peut pas le mettre directement sur le `onClick={useCallback(() => setShowShareModal(true), [setShowShareModal])}`.

Du coup j'ai sorti ça dans deux méthodes séparées. On aurait pu en faire qu'une seule avec un `toggle` qui prendrait le `bool` en prop, mais pour l'utiliser on serait obligé de faire `onClick={() => toggle(bool)}` ce qui revient au même qu'au début...

Ou alors faire un `onClick={toggle(bool)}` avec `const toggle = () => useCallback()` ?

Reply:
For a toggle that is memoized with useCallback, it is helpful to use the function variant of setState:

```
const [show, setShow] = useState(false)
const toggle = useCallback(() => setShow(v => !v), [setShow])
```

Une autre approche, utilisée dans banks : https://github.com/cozy/cozy-banks/blob/master/src/components/useToggle.js